### ISSUE TITLE
Remove dependency to ghc::filesystem?

### ISSUE THREAD

**Issue by gruenich** [doxygen/doxygen#10320]
**Describe the bug**
One dependency of Doxygen is ghc::filesystem which provides a C++17-style std::filesystem class.

**Expected behavior**
Doxygen requires C++-17. We should be able to drop the dependency and use std::filesystem directly.

**Comment by doxygen** [comment]
@albert-github If I remember correctly, at the time of switching to C++17, the GCC version used by GitHub actions did not support std::filesystem out of the box (it needs GCC 9 or higher, earlier versions need special handling as mentioned [here](https://stackoverflow.com/a/68913206/784672)), so that's why I kept the `ghc::filesystem` dependency. Maybe it is time to re-evaluate.

**Comment by albert-github** [comment]
It is always good to reevaluate the usages of extra packages against build in solutions.
I think it still a bit early to remove ghc::filesystem:
- the, potential, problem regarding unicode paths (u8path) but this is probably easily solvable
- the fact that older versions of GCC don't properly support std::filesystem especially GCC 8, where GCC 8.5 has been released on May 14 2021 (see https://gcc.gnu.org/releases.html).

**Comment by gruenich** [comment]
I am not aware of your compatibility policy. GCC 9.1 was released in May 2019, that is more relevant than the last patch release of the GCC 8 series.

**Comment by gruenich** [comment]
Almost two years later, do you still think support for GCC 8 and lower has to be provided?

**Comment by albert-github** [comment]
Yes I do think so, when looking at e.g. Red Hat Linux the GCC version 8 is by default supported by RHEL 8 which enterprise lifetime is somewhere till the mid of 2029, see:
- https://developers.redhat.com/articles/2025/04/16/gcc-and-gcc-toolset-versions-rhel-explainer#
- https://en.wikipedia.org/wiki/Red_Hat_Enterprise_Linux
